addons/core/addon/authenticators/oidc.js

```diff
@@ -0,0 +1,65 @@
+import BaseOIDCAuthenticator from 'auth/authenticators/oidc';
+import { inject as service } from '@ember/service';
+
+/**
+ *
+ */
+export default class OIDCAuthenticator extends BaseOIDCAuthenticator {
+
+  // =services
+
+  @service store;
+
+  // =methods
+
+  /**
+   * Generates an auth method URL with which to authenticate.
+   * @override
+   * @param {object} options
+   * @param {string} options.scope.scope_id
+   * @param {string} options.authMethod.id
+   * @return {string}
+   */
+  buildStartAuthEndpointURL({
+    scope: { id: scopeID },
+    authMethod: { id: authMethodID },
+  }) {
+    const adapter = this.store.adapterFor('application');
+    const options = {
+      adapterOptions: {
+        scopeID,
+        method: 'authenticate:start'
+      }
+    };
+    return adapter.buildURL('auth-method', authMethodID, options, 'findRecord');
+  }
+
+  /**
+   * Generates an auth method URL with which to authenticate.
+   * @override
+   * @param {object} options
+   * @param {string} options.scope.scope_id
+   * @param {string} options.authMethod.id
+   * @return {string}
+   */
+  buildAuthEndpointURL({
+    scope: { id: scopeID },
+    authMethod: { id: authMethodID },
+  }) {
+    const adapter = this.store.adapterFor('application');
+    const options = { adapterOptions: { scopeID, method: 'authenticate' } };
+    return adapter.buildURL('auth-method', authMethodID, options, 'findRecord');
+  }
```

Comment:
Potential for creating a common function that takes `scope`, `authMethod`, and current `adapterOptions.method` string as a param to combine `buildAuthEndpointURL` and `buildStartAuthEndpointURL`. Unless this separation is future proofing for config variations.

Reply:
OIDC has different needs and there is no single "auth" endpoint, so this is now split into `buildStartAuthEndpointURL` and `buildTokenAuthEndpointURL`.